packages/monitor/lib/monitors.js

```diff
@@ -0,0 +1,62 @@
+//
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+'use strict'
+
+const { metrics } = require('./metrics')
+
+function createWriteHead (prevWriteHead, listener) {
+  let fired = false
+
+  // return function with core name and argument list
+  return function writeHead (...args) {
+    const result = prevWriteHead.apply(this, args)
+    if (!fired) {
+      fired = true
+      listener.call(this)
+    }
+
+    return result
+  }
+}
+
+function monitorResponseTimes (defaultLabels = {}) {
+  // see metrics.js for available defaultLabels/to add labels
+  return (req, res, next) => {
+    try {
+      const { method } = req
+      const endTimer = metrics.responseTime.startTimer({ ...defaultLabels, method })
+
+      res.writeHead = createWriteHead(res.writeHead, () => endTimer())
+
+      next()
+    } catch (err) {
+      next(err)
+    }
+  }
+}
+
+function monitorSocketIO (io) {
+  io.on('connection', (socket) => {
+    socket.once('disconnect', () => {
+      metrics.connectionsCount.dec({ type: 'ws' }, 1)
+    })
+    metrics.connectionsCount.inc({ type: 'ws' }, 1)
+    metrics.connectionsTotal.inc({ type: 'ws' }, 1)
+  })
+}
+
+function monitorHttpServer (server) {
+  server.on('connect', () => {
+    metrics.connectionsCount.inc({ type: 'http' }, 1)
+    metrics.connectionsTotal.inc({ type: 'http' }, 1)
+  })
+  server.on('close', () => {
```

Comment:
This is emitted when the server closes. 
https://nodejs.org/api/http.html#event-close_1